project/src/main/webapp/gmail.js

```diff
@@ -15,191 +15,235 @@
-                            const emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex+emailBodyEndIndex);
-                            const emailDateStartIndex = decodedEmail.indexOf("Date:");
-                            const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
-                            var emailDateValue = decodedEmail.substring(emailDateStartIndex, emailDateStartIndex+emailDateEndIndex);
-                            if (emailDateValue.indexOf("X-Notifications") != -1) {
-                                emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
+                        getPageOfMessages(request, result);
+                    } else {
+                        for (var i = 0; i < result.length; i++) {
+                            if (result[i] === undefined) {
+                                console.log("There are no messages from the query");
                             }
-                            var emailObject = {};
-                            emailObject[messageID] = {emailDate: emailDateValue, emailBody: emailBodyValue}; 
-                            if (emailBodyValue.indexOf("format=flowed") == -1 && emailDateValue.length > 1) {
-                                list[list.length] = emailObject;
+                            else {
+                                promiseArray.push(this.getMessage(result[i].id));
                             }
-                        });
-                    }
+                        }
+                        Promise.all(promiseArray).then(() => resolve()
+                        ); //Promise.all and then this.resolve
                     }
                 });
             };
-            console.log(list);
             var initialRequest = gapi.client.gmail.users.messages.list({
-            'userId': 'me',
-            'q': "When is the interview prep workshop?"
+                'userId': 'me',
+                'q': this.question
+            });
+            getPageOfMessages(initialRequest, []);
         });
-        getPageOfMessages(initialRequest, []);
-        const t1 = performance.now();
-        console.log(t1-t0);
+        console.log("The promise ended");
+        return listMessagePromise;
     }
 
-    
-
-/**
- * Get Message with given ID.
- *
- * @param  {String} userId User's email address. The special value 'me'
- * can be used to indicate the authenticated user.
- * @param  {String} messageId ID of Message to get.
- * @param  {Function} callback Function to call when the request is complete.
- */
-function getMessage(userId, messageId, callback) {
-  var request = gapi.client.gmail.users.messages.get({
-    'userId': userId,
-    'id': messageId
-  });
-  request.execute(callback);
-}
-
+    get question() {
+        return this.question;
+    }
+    get totalObjects() {
+        return this.emailObjects;
+    }
```

Comment:
Are these needed? Please add a comment if they are necessary for some reason

Reply:
I removed them in pull request 12. I'll get rid of them here too!